io-engine/src/core/env.rs

```diff
@@ -87,6 +88,38 @@ fn parse_ps_timeout(src: &str) -> Result<Duration, String> {
         .map(|d| d.clamp(Duration::from_secs(1), Duration::from_secs(60)))
 }
 
+/// Parses Command Retry Delay(s): either a single integer or a comma-separated
+/// list of three integers.
+fn parse_crdt(src: &str) -> Result<[u16; 3], String> {
+    fn parse_val(s: &str) -> Result<u16, String> {
+        let u = u16::from_str(s).map_err(|e| e.to_string())?;
+        if u > 100 {
+            Err("Command Retry Delay value is too big".to_string())
+        } else {
+            Ok(u)
+        }
+    }
+
+    let items = src.split(',').collect::<Vec<&str>>();
+    if items.len() == 1 {
+        let u = parse_val(src)?;
+        Ok([u, 0, 0])
+    } else if items.len() == 3 {
+        let mut items = items.into_iter();
+
+        let mut u = [0; 3];
+        for i in &mut u {
+            *i = parse_val(items.next().unwrap())?;
+        }
+
+        Ok(u)
+    } else {
+        Err("Command Retry Delay argument must be a interger or \
+            a comma-separated list of three intergers"
+            .to_string())
+    }
```

Comment:
What about using slices here?

```suggestion
    match items.as_slice() {
        [ one ] => Ok([parse_val(one)?, 0, 0]),
        [ one, two, three ] =>  Ok([parse_val(one)?, parse_val(two)?, parse_val(three)? ]),
        _ => {
            Err("Command Retry Delay argument must be a interger or \
                a comma-separated list of three intergers"
                .to_string())
        }
```

Reply:
ah, it never occurred to me to use match. I'll try it.